in_toto/model.go

```diff
@@ -57,6 +79,42 @@ type Link struct {
 	Environment map[string]interface{} `json:"environment"`
 }
 
+func validateMaterialOrProductHash(value string) error {
```

Comment:
IMO this could be a generic `_validateHexString` function that may be used for keyids, signature value, artifact hashes, etc., whereas...